src/Compilers/CSharp/Test/Syntax/Parsing/PatternParsingTests.cs

```diff
@@ -532,5 +532,379 @@ public void TypeDisambiguation_03()
+            {
+                N(SyntaxKind.IsPatternExpression);
+                {
+                    N(SyntaxKind.IdentifierName);
+                    {
+                        N(SyntaxKind.IdentifierToken, "A");
+                    }
+                    N(SyntaxKind.IsKeyword);
+                    N(SyntaxKind.ConstantPattern);
+                    {
+                        N(SyntaxKind.NullLiteralExpression);
+                        {
+                            N(SyntaxKind.NullKeyword);
+                        }
+                    }
+                }
+                N(SyntaxKind.LessThanToken);
+                N(SyntaxKind.IdentifierName);
+                {
+                    N(SyntaxKind.IdentifierToken, "B");
+                }
+            }
+            EOF();
+        }
+
+        [Fact, WorkItem(15734, "https://github.com/dotnet/roslyn/issues/15734")]
+        public void PatternExpressionPrecedence02b()
+        {
+            UsingExpression("A is B < C");
+            N(SyntaxKind.LessThanExpression);
+            {
+                N(SyntaxKind.IsExpression);
+                {
+                    N(SyntaxKind.IdentifierName);
+                    {
+                        N(SyntaxKind.IdentifierToken, "A");
+                    }
+                    N(SyntaxKind.IsKeyword);
+                    N(SyntaxKind.IdentifierName);
+                    {
+                        N(SyntaxKind.IdentifierToken, "B");
+                    }
+                }
+                N(SyntaxKind.LessThanToken);
+                N(SyntaxKind.IdentifierName);
+                {
+                    N(SyntaxKind.IdentifierToken, "C");
+                }
+            }
+            EOF();
+        }
+
+        [Fact, WorkItem(15734, "https://github.com/dotnet/roslyn/issues/15734")]
+        public void PatternExpressionPrecedence03()
+        {
+            UsingExpression("A is null == B");
+            N(SyntaxKind.EqualsExpression);
+            {
+                N(SyntaxKind.IsPatternExpression);
+                {
+                    N(SyntaxKind.IdentifierName);
+                    {
+                        N(SyntaxKind.IdentifierToken, "A");
+                    }
+                    N(SyntaxKind.IsKeyword);
+                    N(SyntaxKind.ConstantPattern);
+                    {
+                        N(SyntaxKind.NullLiteralExpression);
+                        {
+                            N(SyntaxKind.NullKeyword);
+                        }
+                    }
+                }
+                N(SyntaxKind.EqualsEqualsToken);
+                N(SyntaxKind.IdentifierName);
+                {
+                    N(SyntaxKind.IdentifierToken, "B");
+                }
+            }
+            EOF();
+        }
+
+        [Fact, WorkItem(15734, "https://github.com/dotnet/roslyn/issues/15734")]
+        public void PatternExpressionPrecedence04()
+        {
+            UsingExpression("A is null & B");
```

Comment:
You added tests with operators ==, <, <<, & and &&.
Would it make sense to have tests for other (higher-precedence) operators like + or *?
What about another `is` (although that doesn't seem very useful)?